res/skins/QMLDemo/Mixxx/Controls/Spinny.qml

```diff
@@ -0,0 +1,111 @@
+import Mixxx 0.1 as Mixxx
+import QtQuick 2.12
+
+Item {
+    id: root
+
+    required property string group
+    property int rpm: 33
+    property bool indicatorVisible: true
+    property alias indicatorDelegate: indicatorContainer.data
+
+    function update() {
+        if (!indicatorContainer.visible)
+            return ;
+
+        let positionNormalized = playPositionControl.value;
+        let totalFrames = samplesControl.value / 2;
+        let sampleRate = sampleRateControl.value;
+        if (isNaN(positionNormalized) || isNaN(totalFrames) || isNaN(sampleRate) || totalFrames <= 0 || sampleRate <= 0)
+            return ;
+
+        // Convert playpos to seconds.
+        let t = positionNormalized * totalFrames / sampleRate;
+        // Bad samplerate or number of track samples.
+        if (isNaN(t))
+            return ;
+
+        indicatorRotation.angle = (360 * rpm / 60 * t) % 360;
+    }
+
+    // Avoid animation short blinking of spinny during startup
+    Component.onCompleted: indicatorTransition.enabled = true
+
+    Mixxx.ControlProxy {
+        id: samplesControl
+
+        group: root.group
+        key: "track_samples"
+        onValueChanged: root.update()
+    }
+
+    Mixxx.ControlProxy {
+        id: sampleRateControl
+
+        group: root.group
+        key: "track_samplerate"
+        onValueChanged: root.update()
+    }
+
+    Mixxx.ControlProxy {
+        id: playPositionControl
+
+        group: root.group
+        key: "playposition"
+        onValueChanged: root.update()
```

Comment:
I'm a little bit worried how often this will get called. it would be ideal if it was just once per frame but can we guarantee that?

Reply:
I'll check if we can improve it by using a declarative binding that defines a relationship. I guess QML is then smart enough to figure it out itself.

Reply:
Done.